[ShawnYip-Au] iP

ShawnYip-Au opened the pull request:
# Heading

**bold text**
*italic*
***italic bold text***
~~strike~~

- bullet list 1
- bullet list 2

1. Numbered
2. List

- [ ] task
- [x] list

:satisfied:

[Click for help](https://guides.github.com/features/mastering-markdown/)

> block quote

`in line code`

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
.gitignore: Add src/main/java/*.class

Commit:
Wrap logo into function logo()

Commit:
Implement Duke's introduction

Commit:
.gitignore: Add src/main/java/*.swp

Commit:
Implement Duke asking for inputs

Commit:
Implement Duke end condition upon keyword bye

Commit:
Implement Duke echo non bye inputs

Commit:
Implement store input capabilities

Commit:
Implement view list upon keyword list

Commit:
Implement Task class

Commit:
Add text upon keyword list

Commit:
Implement function markDone() and markUnDone in class Task

Commit:
Implement EventType enum

Commit:
Use enum to detect keyword events

Commit:
Implement mark and unmark event

Commit:
Implement ToDos class which extends Task

Commit:
Override toString() in ToDos

Commit:
Rename ToDos.java => ToDo.java

Commit:
Implement Deadline class which extends Task

Commit:
Implement Event class which extends Task

Commit:
Add TODO, DEADLINE, EVENT into enum EventType

Commit:
Implement todoEvent(), deadlineEvent(), eventEvent()

Commit:
Fix output for Deadline and Event

Commit:
Add total task stored thus far

Commit:
Add Javadocs for removeFirstWord()

Commit:
Update inputs and expected results

Commit:
Implement no arugment exception for ToDo tasks

Commit:
Implement no such keyword exception

Commit:
Code clean up

Commit:
Implement delete event

Commit:
README.md: Add task list

Commit:
Apply IntelliJ suggestions

Commit:
Implement write to local file

Commit:
Create directory and file if it does not exist

Commit:
String format for text file

Commit:
Load from save file

Commit:
Add date format to deadline

Commit:
Add date format to event

Commit:
Create file if it does not exist

Commit:
Fix empty file bug

Commit:
Refine todo

Commit:
Fix todo accept empty input

Commit:
Refine deadline write to file

Commit:
Refine event write to file

Commit:
Implement date format

Commit:
Code clean up

Commit:
Handle IndexOutOfBoundsException for mark, unmark, delete

Commit:
Add ui class

Commit:
Create storage class constructor

Commit:
Implement storage class

Commit:
Implement parse class

Commit:
Implement TaskList

Commit:
Create duke package

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'

Commit:
Add 2 JUnit test

Commit:
Create JAR file

SPWwj commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,59 @@
+    private LocalDate startTime;
+    private LocalDate endTime;
+
+    public Event(String s, String startTime, String endTime) {
```
Perhaps can use a more meaningful name for "s"
```suggestion
    public Event(String s, String startTime, String endTime) {
```

SPWwj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,122 @@
+
+            for (String task : savedTasks) {
+                switch (task.charAt(0)) {
+                    case 'T':
+                        isTaskDone = this.getIsTaskDone(task);
+                        taskDetails = this.getTaskDetails(task);
+                        savedInputs.add(new ToDo(isTaskDone, taskDetails));
+                        break;
+                    case 'D':
+                        isTaskDone = this.getIsTaskDone(task);
+                        taskDetails = this.getTaskDetails(task);
+                        taskDate = this.getTaskDate(task);
+                        savedInputs.add(new Deadline(isTaskDone, taskDetails, taskDate));
+                        break;
+                    case 'E':
+                        isTaskDone = this.getIsTaskDone(task);
+                        taskDetails = this.getTaskDetails(task);
+                        taskDate = this.getTaskDate(task);
+                        savedInputs.add(new Event(isTaskDone, taskDetails, taskDate));
+                        break;
```
Avoid unsafe shortcut, maybe can add in a default case

SPWwj commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,162 @@
+
+        Task newTask;
+        try {
+            String[] infoA = userInput.split("/from");
+            String[] infoB = infoA[1].split("/to");
```
Maybe can use more sensible name instead of A and B :smile:

SPWwj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    private Boolean getIsTaskDone(String s) {
+        String line = s.substring(s.indexOf("|") + 1);
```
Question: why there is a + 1 here?
Maybe can use comment to explain.

SPWwj commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,74 @@
+        System.out.println(taskList);
+    }
+
+    public void printListWithAttitude(TaskList taskList) {
```
Very sensible name.

SPWwj reviewed (commented):
LGTM, almost. Just a few nits to fix.
Overall good quality of code.

harin0826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,84 @@
+
+            // If valid keyword, execute input
+            switch (curEvent) {
+                case BYE:
```
Java coding standard we follow suggest omitting indentation for case clauses. Maybe we can remove those?

harin0826 reviewed (commented):
Nice use of OOP, just minor indentation fixes! :)

Commit:
Try out javadocs

Commit:
Add JavaDocs for class

Commit:
Add JavaDocs for Deadline

Commit:
Add JavaDOcs for DukeExcpetion and Duke

Commit:
Add JavaDOcs for Event

Commit:
Add JavaDocs for Parser

Commit:
Add JavaDocs for Storage

Commit:
Add JavaDocs for public methods and class variables

Commit:
Change boolean variable name

Commit:
Repair naming bug

Commit:
Implement find feature

Commit:
Merge branch 'Level-9'

nusE0726844 reviewed (commented):
I like your use of OOP
Perhaps extracting more classes?

Commit:
Add GUI framework

Commit:
GUI add title

Commit:
Implement duke gui

Commit:
Implement exit gui upon bye

Commit:
Add checkstyle

Commit:
Add settings.gradle to build jar file

Configure gradle to build jar file as intelliJ method is finicky and not transferable.

Commit:
Clean up build.gradle

Remove junk from the file which was included from the tutorial of using gradle.

Commit:
Add assertions

Commit:
Create command package

Commit:
Add javadocs for command package

Commit:
Extract function for command.CommandDeadline

Commit:
Extract function for command.CommandDelete

Commit:
Extract function for command.CommandEvent

Commit:
Extract function for command.CommandFind

Commit:
Extract function for command.CommandList

Commit:
Extract function for command.CommandMark

Commit:
Extract function for command.CommandToDo

Commit:
Extract function for command.CommandUnMark

Commit:
Fix bug in command.CommandToDo

Commit:
Create task package

Commit:
Refractor DukeKeywords to Features

Commit:
Refactor task package

Commit:
Refactor storage package

Commit:
Refactor duke package

Commit:
Change enum keywords to class

Commit:
Fix import statement ordering

Commit:
Refactor Ui class

ShawnYip-Au committed:
Merge pull request #2 from ShawnYip-Au/branch-A-Assertions

Add assertions

Commit:
Merge branch 'master' into A-CodeQuality

Commit:
Re add assertion

ShawnYip-Au committed:
Merge pull request #4 from ShawnYip-Au/branch-A-CodeQuality

Branch a code quality

Commit:
Repair write to storage bug

Commit:
Fix textbox cut content

Commit:
Show user current list

Commit:
Add JavaDocs

Commit:
Implement set priority for task

Commit:
Load priority from file

Commit:
Store priority to file

Commit:
Make profile picture round

Commit:
Scale UI elements

Commit:
Scale dialogue box

Commit:
Tweak responses

Commit:
Fix empty file bug

Commit:
Add Ui.png

Commit:
Remove duplicate README

ShawnYip-Au committed:
Update README.md

Commit:
Grade add dependencies

ShawnYip-Au committed:
Update README.md
